src/v3/src/types/hooks.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { IdxAuthenticator, IdxContext, NextStep } from '@okta/okta-auth-js';
+
+import { HookDefinition as BaseHookDefinition, HookFunction, HookType as BaseHookType } from '../../../types';
+import { FormBag } from './schema';
+import { UserInfo } from './userInfo';
+
+export interface DeviceEnrollment {
+  name: string;
+  platform: string;
+  vendor?: string;
+  signInUrl?: string;
+  enrollmentLink?: string;
+  challengeMethod?: string;
+  orgName?: string;
+}
+
+export interface TransformHookContext {
+  formName: string;
+  currentAuthenticator?: IdxAuthenticator;
+  userInfo?: UserInfo;
+  deviceEnrollment?: DeviceEnrollment;
+  nextStep?: NextStep;
+  idxContext?: IdxContext;
+}
+
+export type TransformHookFunction = (formBag: FormBag, context: TransformHookContext) => void;
```

Comment:
Order of params is inconsistent with the existing event hooks

https://github.com/okta/okta-signin-widget#events

```js
signIn.on('ready', function (context) {/* ... */})
signIn.on('afterError', function (context, error) {/* ... */})
signIn.on('afterRender', function (context) {/* ... */})
```

Reply:
What if we merge context and formBag in one first argument?
```js
signIn.on('afterTransform', function (context) {/* ... */})

context = {
 formBag,
 // context used for existing event hooks
 formName, authenticatorKey, methodType,
 // additional context
 userInfo, deviceEnrollment, nextStep, idxContext,
}
```